src/tests/ftest/nvme/nvme_pool_extend.py

```diff
@@ -136,82 +134,85 @@ def run_ior_thread(self, action, oclass, api, test):
-                                "Pool Version Error:  After extend")
-
-                # Verify the data after pool extend
-                self.run_ior_thread("Read", oclass, api, test)
-                # Get the pool space at the end of the test
-                display_string = "Pool{} space at the End".format(val)
-                self.pool = pool[val]
-                self.pool.display_pool_daos_space(display_string)
-                pool[val].destroy()
-
-                # Stop the extra node servers (rank 6 and 7)
-                output = self.dmg_command.system_stop(self.pool.uuid,
-                                                      "6, 7")
-                self.log.info(output)
-
-    @skipForTicket("DAOS-5869")
+        # The last two available ranks will be used to
+        # extend the created pools
+        initial_extra_ranks = [total_servers - 2, total_servers - 1]
+        # String version to use within dmg
+        extra_ranks = ",".join(map(str, initial_extra_ranks))
+        pver_extend = 0
+
+        for oclass, api, test in product(self.ior_dfs_oclass,
+                                         self.ior_apis,
+                                         self.ior_test_sequence):
+
+            self.pool = self.create_pool()
+            scm_pool_size = self.pool.scm_size
+            nvme_pool_size = self.pool.nvme_size
+
+            self.run_ior_thread("Write", oclass, api, test)
+            self.pool.display_pool_daos_space("Pool space: Beginning")
+
+            pver_begin = self.get_pool_version()
+            self.log.info("Pool Version at the beginning %s", pver_begin)
+
+            # Extend pull, this should modify the pool version
+            output = self.dmg_command.pool_extend(self.pool.uuid,
+                                                  extra_ranks, scm_pool_size,
+                                                  nvme_pool_size)
+            self.log.info(output)
+            # Wait for rebuild to complete
+            self.pool.wait_for_rebuild(to_start=False)
+
+            fail_count = 0
+            while fail_count <= 20:
+                pver_extend = self.get_pool_version()
+                time.sleep(5)
+                fail_count += 1
+                if pver_extend > pver_begin:
+                    break
+            self.log.info("Pool Version after extend %s", pver_extend)
+            # Check pool version incremented after pool extend
+            self.assertTrue(pver_extend > pver_begin,
+                            "Pool Version Error:  After extend")
+
+            # Verify the data after pool extend
+            self.run_ior_thread("Read", oclass, api, test)
+            # Get the pool space at the end of the test
+            display_string = "Pool:{} space at the End".format(self.pool.uuid)
+            self.pool.display_pool_daos_space(display_string)
+
+            # Cleanup
+            self.pool.destroy()
+            time.sleep(10)
```

Comment:
Why do we need sleep here? I think calling pool create immediately after destroy should work. Also, we want to avoid using sleep as much as possible.

Reply:
It should not be needed, just as you suggested. Let me remove it.